src/Microsoft.OData.Core/MultipartMixed/ODataMultipartMixedBatchReader.cs

```diff
@@ -0,0 +1,373 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataMultipartMixedBatchReader.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.MultipartMixed
+{
+    #region Namespaces
+    using System;
+    using System.Diagnostics;
+    using System.Diagnostics.CodeAnalysis;
+    using System.Text;
+
+#if PORTABLELIB
+    using System.Threading.Tasks;
+#endif
+    #endregion Namespaces
+
+    internal sealed class ODataMultipartMixedBatchReader : ODataBatchReader
+    {
+        /// <summary>The batch stream used by the batch reader to divide a batch payload into parts.</summary>
+        private readonly ODataMultipartMixedBatchReaderStream batchStream;
+
+        /// <summary>
+        /// Gets the reader's input context as the real runtime type.
+        /// </summary>
+        private readonly ODataRawInputContext rawInputContext;

```

Comment:
I dislike having the MultipartMixedBatchReader keep track of a separate inputContext from the base class. It doesn't appear that the derived class uses this outside of the constructor, and in a couple of asserts.  Can we remove this private property from the derived class, and make InputContext in the base class protected get if we need to access it from the derived class?

Reply:
moved.

---
In reply to: [145855659](https://github.com/OData/odata.net/pull/960#discussion_r145855659) [](ancestors = 145855659)